feconf.py

```diff
@@ -30,8 +31,9 @@
 # Whether we should serve the development or production experience.
 if PLATFORM == 'gae':
     DEV_MODE = (
-        not os.environ.get('SERVER_SOFTWARE')
-        or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        (not os.environ.get('SERVER_SOFTWARE')
+         or os.environ['SERVER_SOFTWARE'].startswith('Development'))
+        and os.environ.get('MINIFICATION') is None)
```

Comment:
When you log os.environ['MINIFICATION'] here, what do you get?

If it's not the right thing, you might need to investigate where os.environ is being set when an App Engine process starts up.